src/library/baseexternalplaylistmodel.cpp

```diff
@@ -114,23 +105,22 @@ void BaseExternalPlaylistModel::setPlaylist(QString playlist_path) {
     }
 
     QString playlistViewTable = QString("%1_%2").arg(m_playlistTracksTable,
-                                                     QString::number(playlistId));
+            QString::number(playlistId));
 
     QStringList columns;
     columns << "track_id";
     columns << "position";
     columns << "'' AS " + LIBRARYTABLE_PREVIEW;
 
-
     QSqlQuery query(m_database);
     FieldEscaper f(m_database);
     QString queryString = QString(
-        "CREATE TEMPORARY VIEW IF NOT EXISTS %1 AS "
-        "SELECT %2 FROM %3 WHERE playlist_id = %4")
-            .arg(f.escapeString(playlistViewTable),
-                 columns.join(","),
-                 m_playlistTracksTable,
-                 QString::number(playlistId));
+            "CREATE TEMPORARY VIEW IF NOT EXISTS %1 AS "
+            "SELECT %2 FROM %3 WHERE playlist_id = %4")
+                                  .arg(f.escapeString(playlistViewTable),
```

Comment:
I'm unsure if this is ho it should look like

Reply:
Use QStringLiteral to make use of the String builder. And use QSqlQuery::bindValue for the playlist id. It's unlikely that an SQL injection is possible since QString::number is used, but better safe than sorry. And this also prevents people from doing dangerous things when they touch these lines in the future.

Reply:
Hm, I'd prefer not changing any function calls since I am not really familiar with this code.

Reply:
It's not complicated, just replace `%4` with `:playlist_id` and run `query.bindValue(":playlist_id", playlistId)` after `query.prepare(queryString);`

Reply:
Like this? bbabfad11c